### PR TITLE
Rename “compilation_member” as “definition”

### PR DESCRIPTION
**Pull request by paulhuggett**
A long time ago we decided to rename the repo’s “compilation member” as ”definition” to better represent its role in the schema. It’s documented this way on the [prepo glossary page](https://github.com/SNSystems/llvm-project-prepo/wiki/Glossary). Since then I’ve tried to ensure that new code used the new terminology, but there never seemed to be a good time to change the old. This has result in a confusing mishmash of terms that really needs to be resolved.

(See also [llvm-project-prepo PR #134](https://github.com/SNSystems/llvm-project-prepo/pull/134).)

### CODE REVIEW

File: include/pstore/mcrepo/compilation.hpp [hunk 1]
```diff
@@ -81,24 +81,23 @@ namespace pstore {
...
+        //*     _      __ _      _ _   _           *
+        //*  __| |___ / _(_)_ _ (_) |_(_)___ _ _   *
+        //* / _` / -_)  _| | ' \| |  _| / _ \ ' \  *
+        //* \__,_\___|_| |_|_||_|_|\__|_\___/_||_| *
+        //*                                        *
         /// \brief Represents an individual symbol in a compilation.
         ///
-        /// The compilation member provides the connection between a symbol name, its linkage, and
+        /// A definition provides the connection between a symbol name, its linkage, and
         /// the fragment which holds the associated data.
```
[the remark this answers is not in the document]
Comment (paulhuggett) [reply include/pstore/mcrepo/compilation.hpp:92]: It _was_ formatted using clang-format!

File: include/pstore/mcrepo/compilation.hpp [hunk 2]
```diff
@@ -125,23 +124,22 @@ namespace pstore {
                 return static_cast<enum visibility> (visibility_.value ());
             }
 
-            /// \brief Returns a pointer to an in-store compilation member instance.
+            /// \brief Returns a pointer to an in-store definition instance.
             ///
             /// \param db  The database from which the ticket_member should be loaded.
```
[the remark this answers is not in the document]
Comment (paulhuggett) [reply include/pstore/mcrepo/compilation.hpp]: Yes (although that was already wrong) 699813e4da17aaa8faee452849bd5e212289a849

File: include/pstore/mcrepo/compilation.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulhuggett) [reply include/pstore/mcrepo/compilation.hpp]: 4098240840acc833b60877b5539074e7d4cf026a I forced the sentence to be reformatted

**Review by MaggieYingYi — APPROVED** [review]
LGTM!

**Pull request merged**